geometric_kernels/spaces/base.py

```diff
@@ -53,6 +56,23 @@ def get_eigenvalues(self, num: int) -> B.Numeric:
         raise NotImplementedError
 
 
+class DiscreteSpectrumSpaceWithAdditionTheorem(DiscreteSpectrumSpace):
```

Comment:
Why is this needed? To workaround the need specify the exact number of the eigenfunctions as opposed to the number of levels?